File: TAREA 3/1_DESCOMPOSICION_FACTORES_PRIMOS/T3_P1_DESCOMPOSICION_FACTORES_PRIMOS.java
import java.util.Scanner;

public class T3_P1_DESCOMPOSICION_FACTORES_PRIMOS {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese un número entero positivo: ");
        int numero = sc.nextInt();

        System.out.print("Factores primos: ");
        descomponerEnPrimos(numero);
    }

    public static void descomponerEnPrimos(int n) {

        // 1. Extraer todos los factores 2
        while (n % 2 == 0) {
            System.out.print(2 + " ");
            n = n / 2;
        }

        // 2. Extraer factores impares desde 3 hasta √n
        for (int i = 3; i <= Math.sqrt(n); i += 2) {
            while (n % i == 0) {
                System.out.print(i + " ");
                n = n / i;
            }
        }

        // 3. Si queda un número mayor que 2, es primo
        if (n > 2) {
            System.out.print(n);
        }
    }
}

File: TAREA 3/1_DESCOMPOSICION_FACTORES_PRIMOS/T3_P1_DESCOMPOSICION_FACTORES_PRIMOS.py
def factores_primos(n):
    factores = []
    divisor = 2
    while divisor * divisor <= n:
        while n % divisor == 0:
            factores.append(divisor)
            n //= divisor
        divisor += 1
    if n > 1:
        factores.append(n)
    return factores

# Ejemplo
numero = 84
print(f"Factores primos de {numero}: {factores_primos(numero)}")
